ttnn/cpp/ttnn/operations/reduction/generic/device/kernels/compute/reduce_h_interleaved.cpp

```diff
@@ -0,0 +1,47 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cstdint>
+
+#include "compute_kernel_api/reduce.h"
+
+namespace NAMESPACE {
+void MAIN {
+
+    uint32_t Ht = get_compile_time_arg_val(0);
+    uint32_t Wt = get_compile_time_arg_val(1);
+    uint32_t NC = get_compile_time_arg_val(2);
+    uint32_t row_chunk = get_compile_time_arg_val(3);
+
+    reduce_init<true>(tt::CB::c_in0, tt::CB::c_in2);
+    cb_wait_front(tt::CB::c_in2, 1);  // scaler tile from the reader
+
+    constexpr int onetile = 1;
+    for (uint32_t nc = 0; nc < NC; ++nc) {
+        for (uint32_t wt = 0; wt < Wt; wt += row_chunk) {
+            uint32_t chunk_end = std::min(wt + row_chunk, Wt);
+            uint32_t tile_num = std::min(row_chunk, Wt - wt);
+            int reduce_dst_idx = 0;
+
+            // reduction for a chunk of columns
```

Comment:
Please add a long comment about the order the values are coming in, similar to the old kernel, except with more detail.

Need to include details about
- N C W_skip H W_chunk order
- say what the difference is from NCWH order 
- reword original last comment line to something like "in this case we just sequentially add to accumulator all the H-tiles in a column, with each entry of a row chunk uses a different dest index"
With an example (Ht=3, Wt=4, row_chunk=2 is the simplest I could think of), showing wt,ht indices. I'm not sure if this should be triplets (with reduce_dst_index being the third entry, although that seems like too much information).
```
      Wt
Ht 0,0 0,0 1,0 1,0
     0,1. 0,1 1,1 1,1
     0,2. 0,2 1,1 1,1
```

Reply:
Added a detailed comment with the example, hopefully didn't go overboard.